### PR TITLE
[trace] Add trace for Devbox methods

### PR DESCRIPTION
**Pull request by hezhizhen**
## Summary

1. Add `ctx` parameter for most Devbox methods
2. Add trace tasks for Devbox methods
3. Add regions for sub-methods

### CODE REVIEW

**Comment by hezhizhen** [comment]
The PR is still working in progress. Is it a good idea to trace them all? @savil @mikeland86

**Comment by savil** [comment]
@gcurtis do you have thoughts on this? I know you'd added the initial tracing infra, and dashboards.

My concern would be that adding to each method may get overwhelming to track in the dashboards, but i'm not sure where to look for them.

**Comment by hezhizhen** [comment]
> @gcurtis do you have thoughts on this? I know you'd added the initial tracing infra, and dashboards.
> 
> My concern would be that adding to each method may get overwhelming to track in the dashboards, but i'm not sure where to look for them.

Any updates? @savil @gcurtis

**Review by gcurtis — COMMENTED** [review]
This looks good to me. There aren't any dashboards for the CLI, it's just for `devbox -trace` which generates a `go tool trace` file to help track down performance issues.

**Comment by hezhizhen** [comment]
The rest methods are so simple that I don't think it's necessary to add trace. PTAL @gcurtis

**Comment by hezhizhen** [comment]
@gcurtis @savil Any comments?

File: internal/impl/update.go [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // Copyright 2023 Jetpack Technologies Inc and contributors. All rights reserved.
 // Use of this source code is governed by the license in the LICENSE file.
 
+// Copyright 2023 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
```
Comment (gcurtis) [inline internal/impl/update.go:6]: Duplicate?

**Review by gcurtis — APPROVED** [review]
Thanks for cleaning all this up!

**Comment by savil** [comment]
@hezhizhen thanks! if you could resolve the conflcts, I can help merge.

**Comment by hezhizhen** [comment]
> @hezhizhen thanks! if you could resolve the conflcts, I can help merge.

Done. PTAL @savil

File: internal/impl/update.go (hunk 1 above)
Comment (hezhizhen) [reply internal/impl/update.go]: Yeah. That's weird. I removed it.

**Review by savil — APPROVED** [review]

**Pull request merged**